src/filters/utils.js

```diff
@@ -1,50 +1,160 @@
 'use strict';
 
-const groupBy = (objectArray, property) =>
-  objectArray.reduce((acc, obj) => {
-    const key = obj[property];
-    if (!acc[key]) {
-      acc[key] = [];
-    }
-    acc[key].push(obj);
-    return acc;
-  }, {});
+/* @docs
+label: Utility Filters
+category: file
+*/
 
+/* @docs
+label: typeCheck
+category: inspect
+note: Return either the type, or the result of a type-check.
+example: |
+  {{ my_data | typeCheck('array') }}
+params:
+  val:
+    type: any
+    note: The data to check
+  is:
+    type: string
+    default: undefined
+    note: Optinoally check for a particular type
+*/
 const typeCheck = (val, is) => {
   const type = typeof val;
   return is ? type === is : type;
 };
 
-const objectKeys = (obj) => Object.keys(obj);
+/* @docs
+label: get
+category: inspect
+note: |
+  Returns true if an object contains a particular attribute
+  or attr:value pair,
+  including values inside an array.
+example: |
+  {% if page.data | get('author', 'miriam') %}
+params:
+  obj:
+    type: object
+  attrs:
+    type: string | array
+    note: Attribute to filter for
+  value:
+    type: any
+    default: undefined
+    note: Optional value to find within the attribute
+*/
+const get = (obj, attrs, value) => {
+  attrs = typeof attrs === 'string' ? [attrs] : attrs || [];
 
-const slugify = (text) =>
-  text
-    .toString()
-    .toLowerCase()
-    .replace(/\s+/g, '-') // Replace spaces with -
-    .replace(/[^\w-]+/g, '') // Remove all non-word chars
-    .replace(/--+/g, '-') // Replace multiple - with single -
-    .replace(/^-+/, '') // Trim - from start of text
-    .replace(/-+$/, ''); // Trim - from end of text
+  if (attrs && attrs.length) {
```

Comment:
Above, on line 49, I re-assign it so we can pass in a string, but always count on getting an  array. Is there a better way to handle that? We're also re-assigning `obj` inside the if statement.

(I know very little about JS or best-practice here, so I need your input)

Reply:
I think in general it's not a good practice to assign function arguments since its unclear to the caller that they'll be modified and can lead to unexpected behavior.  We should probably also include a default value for `value` if it's an optional argument.

Reply:
ok, I can take a stab at this, but it's another place I don't quite understand JS yet, so we'll see :p